swagger.yaml

```
@@ -10,33 +10,322 @@ basePath: "/v1"
+        400:
+          description: "Invalid request"
+        404:
+          $ref: "#/responses/NotFound"
+        500:
+          $ref: "#/responses/InternalError"
+    put:
+      tags:
+        - "policies"
+      summary: "Updates a policy"
+      description: "Updates a policy for a given id"
       produces:
-        - application/json
+        - "application/json"
+      parameters:
+        - in: path
+          name: id
+          description: "Unique id of policy"
+          type: string
+          required: true
       responses:
         200:
-          description: OK
+          description: "Successfully updated a policy for a given id"
           schema:
-            $ref: "#/definitions/HelloResponse"
+            $ref: "#/definitions/Policy"
+        400:
+          description: "Invalid request"
+        404:
+          $ref: "#/responses/NotFound"
+        500:
+          $ref: "#/responses/InternalError"
+
+  /authorise:
+    post:
+      tags: 
+        - "permissions"
+      summary: "Returns status code indicating if the user is authorised"
+      description: "Returns a status code for a specifc user. The user details, user_id, group_id and roles are passed in by a JWT. The status code indicates if the user is authorised for this role."
+      produces:
+        - "application/json"
+      security:
+        - UserAuthorisationToken: []
+      responses:
+        200:
+          description: "User is authorised for this role"
+        400:
+          description: "Invalid request"
+        401:
+          description: "Unauthorised request"
+        403:
+          description: "User is unauthorised for this role"
         500:
-          $ref: '#/responses/InternalError'
+          $ref: "#/responses/InternalError"
+
+
 
 responses:
   InternalError:
     description: "Failed to process the request due to an internal error"
+  NotFound:
+    description: "Requested id can not be found"
 
-definitions:
-  HelloResponse:
+definitions: 
+  Role:
     type: object
+    required:
+      - id
+      - role
+      - permissions
     properties:
-      message:
+      id:
+        $ref: "#/definitions/RoleId"
+      role:
+        description: "Name of role"
+        type: string
+        example: "admin"
+      permissions:
+        description: "A list of permissions associated with this role"
+        type: array
+        items:
+          $ref: "#/definitions/Permission"
+        example: "read only"
+  Permission:
+    type: object
+    required:
+      - id
+      - description
+    properties:
+      id:
+        description: "Unique id for a permission"
+        type: string
+        example: "1"
+      description:
+        description: "Description for a permission"
+        type: string
+        example: "read only"
+  RoleId:
+    type: object
+    required:
+      - id
+    properties:
+      id:
+        description: "Unique id for a role"
+        type: string
+        example: "1"
```

Is there a need for the RoleId to be an object? If not:
```suggestion
  RoleId:
        description: "Unique id for a role"
        type: string
        example: "1"
```

I do get the feeling this might be about me not understanding the requirements though 🙈

---

I will update this, this is a good idea